spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteUpdate.scala

```diff
@@ -65,7 +100,7 @@ case class RewriteUpdate(spark: SparkSession) extends Rule[LogicalPlan] with Rew
       relation: DataSourceV2Relation,
       scanPlan: LogicalPlan,
       assignments: Seq[Assignment],
-      cond: Expression): LogicalPlan = {
+      cond: Expression = Literal.TrueLiteral): LogicalPlan = {
```

Comment:
When we process matched row, we know all rows must be updated. No need for `If` expressions.